test/node-examples.js

```diff
@@ -0,0 +1,132 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Enables ES2015 import/export in Node.js
+
+const {execFile, execFileSync} = require('child_process');
+const puppeteer = require('puppeteer');
+const console = require('console');
+const process = require('process');
+
+const fs = require('fs');
+const PNG = require('pngjs').PNG;
+const pixelmatch = require('pixelmatch');
+
+function printResult(diffRatio, threshold) {
+  return diffRatio <= threshold
+    ? console.log('\x1b[32m%s\x1b[0m', 'Rendering test Passed!')
+    : console.log('\x1b[31m%s\x1b[0m', 'Rendering test failed!');
+}
+
+async function validateWithWaitingTime(child, folder, waitingTime, threshold) {
+  const browser = await puppeteer.launch({
+    headless: false,
+    executablePath: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
+  });
+  const page = await browser.newPage();
+  await page.waitFor(2000);
+  await page.goto('http://localhost:8080');
+  await page.setViewport({width: 1000, height: 800});
+  await page.waitFor(waitingTime);
+  await page.screenshot({path: 'new.png'});
+
+  const goldImageData = fs.readFileSync(`../test/golden-images/${folder}.png`);
+  const goldImage = PNG.sync.read(goldImageData);
+  const newImageData = fs.readFileSync('new.png');
+  const newImage = PNG.sync.read(newImageData);
+  const diffImage = new PNG({width: goldImage.width, height: goldImage.height});
+
+  const pixelDiffSize = pixelmatch(
+    goldImage.data,
+    newImage.data,
+    diffImage.data,
+    goldImage.width,
+    goldImage.height,
+    {threshold: 0.105, includeAA: true}
+  );
+  const pixelDiffRatio = pixelDiffSize / (goldImage.width * goldImage.height);
+
+  console.log(`Mismatched pixel number: ${pixelDiffSize}`);
+  console.log(`Mismatched pixel ratio: ${pixelDiffRatio}`);
+
+  const diffImageData = PNG.sync.write(diffImage, {colorType: 6});
+  fs.writeFileSync('diff.png', diffImageData);
+  fs.unlinkSync('new.png');
+  fs.unlinkSync('diff.png');
+
+  child.kill();
+  await page.waitFor(1000);
+  await browser.close();
+  printResult(pixelDiffRatio, threshold);
+  return pixelDiffRatio <= threshold;
+}
+
+async function yarnAndLaunchWebpack() {
+  const output = execFileSync('yarn'); //eslint-disable-line
+  // console.log(output.toString('utf8'));
+  const child = execFile(
+    './node_modules/.bin/webpack-dev-server',
+    ['--env.local', '--config', 'webpack.config.js', '--progress', '--hot'],
+    {
+      maxBuffer: 5000 * 1024
+    },
+    err => {
+      if (err) {
+        console.error(err);
+        return;
+      }
+    }
+  );
+  return child;
+}
+
+function changeFolder(folder) {
+  console.log('--------------------------');
+  console.log(`Begin to test ${folder}`);
+  process.chdir(`./${folder}`);
+}
+
+async function TestExample(folder) {
```

Comment:
We use PascalCase for classes, camelCase for functions. Rename this function to `runTestExample`?